lib/azure/armrest/configuration.rb

```diff
@@ -202,7 +202,16 @@ def token_expiration
       # argument to 'USGov'. Otherwise, set it to nil.
       #
       def environment=(env)
-        fetch_token if env != environment
+        if env != environment
+          if env.to_s.casecmp(Azure::Armrest::USGOV_ENVIRONMENT) == 0
```

Comment:
Better use `case when` statement so in the future we can easily add other environments without lots of if/else